packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -454,74 +530,66 @@ class FlutterLocalFileComparator extends FlutterGoldenFileComparator with LocalC
-    baseDirectory ??= FlutterGoldenFileComparator.getBaseDirectory(
-      defaultComparator,
-      platform,
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+      localFileComparator,
+      platform: platform,
+      fs: fs,
     );
-
     if (!baseDirectory.existsSync()) {
       baseDirectory.createSync(recursive: true);
     }
 
-    goldens ??= SkiaGoldClient(baseDirectory);
-    try {
-      // Check if we can reach Gold.
-      await goldens.getExpectationForTest('');
-    } on io.OSError catch (_) {
-      return FlutterSkippingFileComparator(
-        baseDirectory.uri,
-        goldens,
-        'OSError occurred, could not reach Gold. '
-          'Switching to FlutterSkippingGoldenFileComparator.',
-      );
-    } on io.SocketException catch (_) {
-      return FlutterSkippingFileComparator(
-        baseDirectory.uri,
-        goldens,
-        'SocketException occurred, could not reach Gold. '
-          'Switching to FlutterSkippingGoldenFileComparator.',
-      );
-    }
-
-    return FlutterLocalFileComparator(baseDirectory.uri, goldens);
+    final SkiaGoldClient goldens = SkiaGoldClient(
+      baseDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: httpClient,
+      log: log,
+    );
+    return FlutterLocalFileComparator(
+      baseDirectory.uri,
+      goldens,
+      fs: fs,
+      platform: platform,
+      log: log,
+    );
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
     golden = _addPrefix(golden);
     final String testName = skiaClient.cleanTestName(golden.path);
     late String? testExpectation;
-    testExpectation = await skiaClient.getExpectationForTest(testName);
-
-    if (testExpectation == null || testExpectation.isEmpty) {
-      // There is no baseline for this test.
-      // Ideally we would use markTestSkipped here but in some situations,
-      // comparators are called outside of tests.
-      // See also: https://github.com/flutter/flutter/issues/91285
-      // ignore: avoid_print
-      print(
-        'No expectations provided by Skia Gold for test: $golden. '
-        'This may be a new test. If this is an unexpected result, check '
-        'https://flutter-gold.skia.org.\n'
-        'Validate image output found at $basedir'
-      );
-      update(golden, imageBytes);
+    try {
+      testExpectation = await skiaClient.getExpectationForTest(testName, retryOnFailure: false);
+      if (testExpectation == null || testExpectation.isEmpty) {
+        log(
+          'No expectations provided by Skia Gold for test: $golden. '
+          'This may be a new test. If this is an unexpected result, check '
+          'https://flutter-gold.skia.org.\n'
+          'Validate image output found at $basedir'
+        );
+        update(golden, imageBytes);
+        return true;
+      }
+    } on io.SocketException {
```

Comment:
I couldn't figure out when an OSError would fire (SocketException wraps OSError normally for network stuff?), but I'll add it back to be sure.

Reply:
honestly i mostly removed this because i didn't like the code duplication, but i used matan's trick to avoid code duplication (catch all exceptions, rethrow if it's not the ones we want)

Reply:
Perfect!